### PR TITLE
PR-BAEL-3379

### PR DESCRIPTION
**Pull request by yavuztas**
Add source code for article BAEL-3379

### CODE REVIEW

**Comment by eric-martin** [comment]
Jenkins - retest this please

File: javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        ValidatorFactory validatorFactory = Validation.byDefaultProvider()
+          .configure()
+          .messageInterpolator(new ParameterMessageInterpolator())
+          .buildValidatorFactory();
+
+        validator = validatorFactory.getValidator();
+    }
+
+    @Test
+    public void ifNameLengthIsLess_nameValidationFails() {
```
Comment (KevinGilmore) [inline javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java]:
```suggestion
    public void givenNameLengthLessThanMin_whenValidate_thenValidationFails() {
```

File: javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        Set<ConstraintViolation<Person>> violations = validator.validate(person);
+        assertEquals(1, violations.size());
+
+        ConstraintViolation<Person> violation = violations.iterator().next();
+        assertEquals("Name should be between 10 and 100 characters", violation.getMessage());
+    }
+
+    @Test
+    public void ifAgeIsLess_ageMinValidationFails() {
```
Comment (KevinGilmore) [inline javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java]:
```suggestion
    public void givenAgeIsLessThanMin_whenValidate_thenValidationFails() {
```

File: javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        Set<ConstraintViolation<Person>> violations = validator.validate(person);
+        assertEquals(1, violations.size());
+
+        ConstraintViolation<Person> violation = violations.iterator().next();
+        assertEquals("Age should not be less than 18", violation.getMessage());
+    }
+    
+    @Test
+    public void ifEmailIsMalformed_emailFormatValidationFails() {
```
Comment (KevinGilmore) [inline javaxval/src/test/java/org/baeldung/javaxval/messageinterpolator/ParameterMessageInterpolaterIntegrationTest.java]:
```suggestion
    public void givenEmailIsMalformed_whenValidate_thenValidationFails() {
```

**Pull request merged**